cmd/update.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const githubSlug = "screwdriver-cd/sd-local"
+
+var (
+	currentVersion = version
+	updateFlag     = false
+)
+
+func canUpdate() (*selfupdate.Release, error) {
+
+	if currentVersion == "dev" {
+		return &selfupdate.Release{}, errors.New("This is a development version and cannot be updated")
+	}
+
+	latest, found, err := selfupdate.DetectLatest(githubSlug)
+	if err != nil {
+		return &selfupdate.Release{}, err
+	}
+	if !found {
+		return &selfupdate.Release{}, errors.New("Repositry Not Found")
+	}
+	v := semver.MustParse(currentVersion)
+
+	if latest.Version.LTE(v) {
+		return &selfupdate.Release{}, errors.New("Current version is latest")
+	}
+
+	return latest, nil
+}
+
+func checkUserInput(input string) (bool, error) {
```

Comment:
[imo]
Looking at the function name, it doesn't feel that the return value boolean is an aborted flag.
Is it better to use named return value or change the function name?